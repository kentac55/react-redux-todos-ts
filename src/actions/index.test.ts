import * as actions from '.'

const todo = { id: 0, text: 'text', completed: false }

it('should create a started action to add a todo', () => {
  expect(actions.addTodo.started({ text: 'text' })).toMatchSnapshot()
})

it('should create a done action to add a todo', () => {
  expect(
    actions.addTodo.done({
      params: { text: 'text' },
      result: {
        type: 'Ok',
        contents: todo,
      },
    })
  ).toMatchSnapshot()
})

it('should create a failed action to add a todo', () => {
  expect(
    actions.addTodo.failed({
      params: { text: 'text' },
      error: Error('Error'),
    })
  ).toMatchSnapshot()
})

it('should create a started action to toggle a todo', () => {
  expect(actions.toggleTodo.started({ todo })).toMatchSnapshot()
})

it('should create a done action to toggle a todo', () => {
  expect(
    actions.toggleTodo.done({
      params: { todo },
      result: {
        type: 'Ok',
        contents: { ...todo, completed: !todo.completed },
      },
    })
  ).toMatchSnapshot()
})

it('should create a failed action to toggle a todo', () => {
  expect(
    actions.toggleTodo.failed({
      params: { todo },
      error: Error('Error'),
    })
  ).toMatchSnapshot()
})

it('should create an action to set SHOW_ALL filter', () => {
  expect(actions.setVisibilityFilter('SHOW_ALL')).toMatchSnapshot()
})

it('should create an action to set SHOW_ACTIVE filter', () => {
  expect(actions.setVisibilityFilter('SHOW_ACTIVE')).toMatchSnapshot()
})

it('should create an action to set SHOW_COMPLETED filter', () => {
  expect(actions.setVisibilityFilter('SHOW_COMPLETED')).toMatchSnapshot()
})

it('should create a started action to get a todo', () => {
  expect(actions.getTodo.started({ id: 0 })).toMatchSnapshot()
})

it('should create a done action to get a todo', () => {
  expect(
    actions.getTodo.done({
      params: { id: 0 },
      result: {
        type: 'Ok',
        contents: todo,
      },
    })
  ).toMatchSnapshot()
})

it('should create a failed action to get a todo', () => {
  expect(
    actions.getTodo.failed({
      params: { id: 0 },
      error: Error('Error'),
    })
  ).toMatchSnapshot()
})

it('should create a started action to get todos', () => {
  expect(actions.getTodos.started()).toMatchSnapshot()
})

it('should create a done action to get todos', () => {
  expect(
    actions.getTodos.done({
      params: undefined,
      result: {
        type: 'Ok',
        contents: [todo],
      },
    })
  ).toMatchSnapshot()
})

it('should create a failed action to get todos', () => {
  expect(
    actions.getTodos.failed({
      params: undefined,
      error: Error('Error'),
    })
  ).toMatchSnapshot()
})

it('should create a started action to init app', () => {
  expect(actions.initApp.started()).toMatchSnapshot()
})

it('should create a done action to init app', () => {
  expect(
    actions.initApp.done({
      params: undefined,
      result: {
        type: 'Ok',
        contents: [todo],
      },
    })
  ).toMatchSnapshot()
})

it('should create a failed action to init app', () => {
  expect(
    actions.initApp.failed({
      params: undefined,
      error: Error('Error'),
    })
  ).toMatchSnapshot()
})
