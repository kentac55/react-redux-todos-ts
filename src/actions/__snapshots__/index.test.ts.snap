// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create a done action to add a todo 1`] = `
Object {
  "payload": Object {
    "params": Object {
      "text": "text",
    },
    "result": Object {
      "contents": Object {
        "completed": false,
        "id": 0,
        "text": "text",
      },
      "type": "Ok",
    },
  },
  "type": "AddTodo_DONE",
}
`;

exports[`should create a done action to get a todo 1`] = `
Object {
  "payload": Object {
    "params": Object {
      "id": 0,
    },
    "result": Object {
      "contents": Object {
        "completed": false,
        "id": 0,
        "text": "text",
      },
      "type": "Ok",
    },
  },
  "type": "GetTodo_DONE",
}
`;

exports[`should create a done action to get todos 1`] = `
Object {
  "payload": Object {
    "params": undefined,
    "result": Object {
      "contents": Array [
        Object {
          "completed": false,
          "id": 0,
          "text": "text",
        },
      ],
      "type": "Ok",
    },
  },
  "type": "GetTodos_DONE",
}
`;

exports[`should create a done action to init app 1`] = `
Object {
  "payload": Object {
    "params": undefined,
    "result": Object {
      "contents": Array [
        Object {
          "completed": false,
          "id": 0,
          "text": "text",
        },
      ],
      "type": "Ok",
    },
  },
  "type": "InitApp_DONE",
}
`;

exports[`should create a done action to toggle a todo 1`] = `
Object {
  "payload": Object {
    "params": Object {
      "todo": Object {
        "completed": false,
        "id": 0,
        "text": "text",
      },
    },
    "result": Object {
      "contents": Object {
        "completed": true,
        "id": 0,
        "text": "text",
      },
      "type": "Ok",
    },
  },
  "type": "ToggleTodo_DONE",
}
`;

exports[`should create a failed action to add a todo 1`] = `
Object {
  "error": true,
  "payload": Object {
    "error": [Error: Error],
    "params": Object {
      "text": "text",
    },
  },
  "type": "AddTodo_FAILED",
}
`;

exports[`should create a failed action to get a todo 1`] = `
Object {
  "error": true,
  "payload": Object {
    "error": [Error: Error],
    "params": Object {
      "id": 0,
    },
  },
  "type": "GetTodo_FAILED",
}
`;

exports[`should create a failed action to get todos 1`] = `
Object {
  "error": true,
  "payload": Object {
    "error": [Error: Error],
    "params": undefined,
  },
  "type": "GetTodos_FAILED",
}
`;

exports[`should create a failed action to init app 1`] = `
Object {
  "error": true,
  "payload": Object {
    "error": [Error: Error],
    "params": undefined,
  },
  "type": "InitApp_FAILED",
}
`;

exports[`should create a failed action to toggle a todo 1`] = `
Object {
  "error": true,
  "payload": Object {
    "error": [Error: Error],
    "params": Object {
      "todo": Object {
        "completed": false,
        "id": 0,
        "text": "text",
      },
    },
  },
  "type": "ToggleTodo_FAILED",
}
`;

exports[`should create a started action to add a todo 1`] = `
Object {
  "payload": Object {
    "text": "text",
  },
  "type": "AddTodo_STARTED",
}
`;

exports[`should create a started action to get a todo 1`] = `
Object {
  "payload": Object {
    "id": 0,
  },
  "type": "GetTodo_STARTED",
}
`;

exports[`should create a started action to get todos 1`] = `
Object {
  "payload": undefined,
  "type": "GetTodos_STARTED",
}
`;

exports[`should create a started action to init app 1`] = `
Object {
  "payload": undefined,
  "type": "InitApp_STARTED",
}
`;

exports[`should create a started action to toggle a todo 1`] = `
Object {
  "payload": Object {
    "todo": Object {
      "completed": false,
      "id": 0,
      "text": "text",
    },
  },
  "type": "ToggleTodo_STARTED",
}
`;

exports[`should create an action to set SHOW_ACTIVE filter 1`] = `
Object {
  "payload": "SHOW_ACTIVE",
  "type": "SetVisibilityFilter",
}
`;

exports[`should create an action to set SHOW_ALL filter 1`] = `
Object {
  "payload": "SHOW_ALL",
  "type": "SetVisibilityFilter",
}
`;

exports[`should create an action to set SHOW_COMPLETED filter 1`] = `
Object {
  "payload": "SHOW_COMPLETED",
  "type": "SetVisibilityFilter",
}
`;
